### PR TITLE
Replace coveralls with codecov

### PR DESCRIPTION
**Pull request by willbarton**
This PR switches us from using coveralls to Codecov.io. Codecov has a nicer UI that is easier to browse and more configurable options (configured in a yml file like Travis) than coveralls provides.

As you can see on this PR, there are two coverage tests, "Project" and "Patch". "Project" is an overall measurement of the code coverage for the project compared against the base of the pull request. "Patch" is a comparison of the PR branch to the base branch. I'm not convinced we need both — I think "patch" will suffice for our needs. However, we might want to see what it looks like for a while before making that decision. 

Additionally, "Project" doesn't work right now because Codecov thinks our default branch is flapjack. I've got an open question to Codecov about this, but I suspect that might be fixed when this is merged and `branch: master` in the codecov.yml file is in master. 

Right now I've also set the [coverage drop threshold](https://docs.codecov.io/v4.3.6/docs/commit-status#section-threshold) to 0% drop for both project and patch coverage.

## Additions

- Codecov support for code coverage

## Removals

- Coveralls support

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: gulp/tasks/test.js [hunk 1]
```diff
@@ -406,18 +406,18 @@ function spawnProtractor( ) {
...
-    .pipe( gulpCoveralls() );
+    .pipe( gulpCodecov() );
 }
 
 gulp.task( 'test', [ 'lint', 'test:unit' ] );
 gulp.task( 'test:a11y', testA11y );
 gulp.task( 'test:acceptance', testAcceptanceBrowser );
 gulp.task( 'test:acceptance:protractor', spawnProtractor );
-gulp.task( 'test:coveralls', testCoveralls );
+gulp.task( 'test:codecov', testCodecov );
```
Comment (anselmbradford) [inline gulp/tasks/test.js]: Is there a tool-agnostic name we could give this task? `gulp test:cloud` perhaps? Any other ideas?

File: gulp/tasks/test.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const configTest = require( '../config' ).test;
 const envvars = require( '../../config/environment' ).envvars;
 const fsHelper = require( '../utils/fs-helper' );
 const gulp = require( 'gulp' );
-const gulpCoveralls = require( 'gulp-coveralls' );
+const gulpCodecov = require('gulp-codecov');
```
Comment (anselmbradford) [inline gulp/tasks/test.js]: Add spaces inside the parentheses.

Comment (anselmbradford) [reply gulp/tasks/test.js]: Can be fixed with `gulp lint --fix`

File: gulp/tasks/test.js (hunk 1 above)
Comment (willbarton) [reply gulp/tasks/test.js]: `test:submit-coverage`?

**Comment by anselmbradford** [comment]
> Codecov thinks our default branch is flapjack. 

I remember another service having this same issue, maybe Coveralls actually!
Are you sure it's not because the org permissions aren't set correctly and the most recent version of the site codecov has is from 18 months ago or whatever.

**Comment by willbarton** [comment]
As far as  I can tell, all org permissions are correct, and I just added the repo to codecov yesterday.

I'm inclined to get it in, submit coverage for master, and see what happens from there.

File: gulp/tasks/test.js (hunk 1 above)
Comment (anselmbradford) [reply gulp/tasks/test.js]: It's kiiiiinda long

Comment (willbarton) [reply gulp/tasks/test.js]: But nice and descriptive!

**Comment by willbarton** [comment]
@higs4281 I'll be happy to do so once I know what needs to be done. For right now I've simply added the repo to codecov using the "Add new repository" button. Everything else is in [codecov.yml](https://github.com/cfpb/cfgov-refresh/blob/codecov/codecov.yml).

**Pull request merged**